packages/aws-cdk-lib/aws-appsync/lib/eventapi.ts

```diff
@@ -0,0 +1,584 @@
+   */
+  public readonly logGroup: ILogGroup;
+
+  private api: CfnApi;
+  private eventConfig: CfnApi.EventConfigProperty;
+  private domainNameResource?: CfnDomainName;
+
+  constructor(scope: Construct, id: string, props: EventApiProps) {
+    if (props.apiName !== undefined && !Token.isUnresolved(props.apiName)) {
+      if (props.apiName.length < 1 || props.apiName.length > 50) {
+        throw new Error(`\`apiName\` must be between 1 and 50 characters, got: ${props.apiName.length} characters.`);
+      }
+
+      const apiNamePattern = /^[A-Za-z0-9_\-\ ]+$/;
+
+      if (!apiNamePattern.test(props.apiName)) {
+        throw new Error(`\`apiName\` must contain only alphanumeric characters, underscores, hyphens, and spaces, got: ${props.apiName}`);
+      }
+    }
+
+    super(scope, id, {
+      physicalName: props.apiName ?? Lazy.string({
+        produce: () =>
+          Names.uniqueResourceName(this, {
+            maxLength: 50,
+            separator: '-',
+          }),
+      }),
+    });
+
+    this.validateOwnerContact(props.ownerContact);
+
+    this.authProviderTypes = this.setupAuthProviderTypes(props.authorizationConfig?.authProviders);
+
+    const authProviders = props.authorizationConfig?.authProviders ?? [{ authorizationType: AuthorizationType.API_KEY }];
+    this.validateAuthorizationProps(authProviders);
+
+    const connectionAuthModeType = props.authorizationConfig?.connectionAuthModeTypes ?? this.authProviderTypes;
+    const defaultPublishAuthModeTypes = props.authorizationConfig?.defaultPublishAuthModeTypes ?? this.authProviderTypes;
+    const defaultSubscribeAuthModeTypes = props.authorizationConfig?.defaultSubscribeAuthModeTypes ?? this.authProviderTypes;
+
+    this.validateAuthorizationConfig(authProviders, connectionAuthModeType);
+    this.validateAuthorizationConfig(authProviders, defaultPublishAuthModeTypes);
+    this.validateAuthorizationConfig(authProviders, defaultSubscribeAuthModeTypes);
+
+    this.defaultPublishModeTypes = defaultPublishAuthModeTypes;
+    this.defaultSubscribeModeTypes = defaultSubscribeAuthModeTypes;
+
+    if (!Token.isUnresolved(props.ownerContact) && props.ownerContact !== undefined && props.ownerContact.length > 256) {
+      throw new Error('You must specify `ownerContact` as a string of 256 characters or less.');
+    }
+
+    this.eventConfig = {
+      authProviders: this.mapAuthorizationProviders(authProviders),
+      connectionAuthModes: this.mapAuthorizationConfig(connectionAuthModeType),
+      defaultPublishAuthModes: this.mapAuthorizationConfig(defaultPublishAuthModeTypes),
+      defaultSubscribeAuthModes: this.mapAuthorizationConfig(defaultSubscribeAuthModeTypes),
+      logConfig: this.setupLogConfig(props.logConfig),
+    };
+
+    this.api = new CfnApi(this, 'Resource', {
+      name: this.physicalName,
+      ownerContact: props.ownerContact,
+      eventConfig: this.eventConfig,
+    });
+
+    this.api.applyRemovalPolicy(RemovalPolicy.DESTROY);
```

Comment:
For my knowledge, is this the intended & cloudformation default for event api?

Reply:
This should be the default and doesn't need to be called out explicitly in code. Will remove.